gremlin-javascript/src/main/javascript/gremlin-javascript/lib/driver/connection.js

```diff
@@ -173,7 +173,8 @@ class Connection extends EventEmitter {
 
   /** @override */
   submit(processor, op, args, requestId) {
-    const rid = requestId || utils.getUuid();
+    // TINKERPOP-2847: Use lower case to prevent string comparison issues.
+    const rid = (requestId || utils.getUuid()).toLowerCase();
```

Comment:
you might think adding new function to `utils` which will lowercase requestId if present or build new one.
Or keep as is

Reply:
I think I'll leave it for now. This is a good point that you made but I generally wait until three usages of something similar before extracting it out into its own function.